### ISSUE TITLE
Issues found after trying to generate big swagger file

### ISSUE THREAD

**Issue by szotp** [epam-cross-platform-lab/swagger-dart-code-generator#58]
Hi, I like the idea of dart only code generator, but unfortunately, I found a lot of issues while trying to generate code for my API:

1. additionalProperties key from schema is not handled, I'm getting `final null foo;` variables. At least it should use `final null foo;`.
2. Responses that return bool don't work, it will generate `Future<Response<Bool>>`. It seems that only `num` primitive is handled properly.
3. Models with strange names are not supported, I had few named Foo-Bar. Model name was ok, But parameter name in request method was not.
4. Keywords are not handled in all places, I had enum cases named default or new.
5. The code quality is not so good, I'm seeing a lot of duplicate code that creates the mismatch. 
6. Generated enum name may conflict with generated class name.
7. I don't see any benefit of chopper here, you could generate very similiar code to what chopper does and avoid unnecessary dependency.

**Comment by fryette** [comment]
Could you please share swagger.json?
Also can send it to the ysampir@gmail.com

**Comment by Vovanella95** [comment]
Hi @szotp,

About point 2.

To have Future<bool> at response, you need to have "boolean" return type in response. I guess you have "bool" now. But according spec there is no "bool" base type https://swagger.io/specification/. There is only "boolean".

Please correct me if I wrong.

**Comment by szotp** [comment]
Hi, thanks for feedback, I will prepare minimal swagger spec to illustrate the problems.

**Comment by fryette** [comment]
@szotp any progress?

**Comment by szotp** [comment]
Sorry I had some work to do during the weekend, I'll do it this week.

**Comment by Vovanella95** [comment]
@szotp Hey,

Points 3, 4 and 6  are already fixed!

**Comment by szotp** [comment]
Here is the json, some issues still remain. Good luck!

```
{
    "swagger": "2.0",
    "info": {
        "version": "1.0.0",
        "title": "Swagger Petstore",
        "description": "A sample API that uses a petstore as an example to demonstrate features in the swagger-2.0 specification",
        "termsOfService": "http://swagger.io/terms/",
        "contact": {
            "name": "Swagger API Team"
        },
        "license": {
            "name": "MIT"
        }
    },
    "host": "petstore.swagger.io",
    "basePath": "/api",
    "schemes": [
        "http"
    ],
    "consumes": [
        "application/json"
    ],
    "produces": [
        "application/json"
    ],
    "paths": {
        "/pets": {
            "get": {
                "description": "Returns all pets from the system that the user has access to",
                "produces": [
                    "application/json"
                ],
                "parameters": [
                    {
                        "in": "body",
                        "name": "foobar",
                        "required": true,
                        "schema": {
                            "$ref": "#/definitions/foo-Bar"
                        }
                    }
                ],
                "responses": {
                    "200": {
                        "responses": {
                            "200": {
                                "description": "No response was specified",
                                "schema": {
                                    "type": "boolean"
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    "definitions": {
        "Response": {
            "type": "object",
            "required": [],
            "properties": {
                "preferences": {
                    "additionalProperties": {
                        "type": "string"
                    }
                },
                "foo": {
                    "type": "string",
                    "enum": [
                        "1ENUM",
                        "2$@!enum",
                        "new",
                        "default"
                    ]
                }
            }
        },
        "ResponseFoo": {
            "type": "object"
        },
        "foo-Bar": {
            "type": "object"
        }
    }
}
```

**Comment by Vovanella95** [comment]
Hi @szotp , We fixed all issues in 1.2.0 version. Refactoring issue is moved to separate one. In future we're going to implement native requests (not using chopper), but not right now :)
